politeiawww/user/user.go

```diff
@@ -455,8 +455,43 @@ func DecodeSession(payload []byte) (*Session, error) {
 	return &s, nil
 }
 
+// EmailHistory24h represents a 24h history for a User identified by email.
+type EmailHistory24h struct {
+	// Email is an email address for some User this history belongs to.
+	Email string `json:"email"`
```

Comment:
Well, probably yes, but then how are we gonna identify the same DB entry when trying to update it instead of creating from scratch ?

According to the following comment from `gorm`:
```go
// Save update value in database, if the value doesn't have primary key, will insert it
func (s *DB) Save(value interface{}) *DB {
	scope := s.NewScope(value)
	if !scope.PrimaryKeyZero() {
		newDB := scope.callCallbacks(s.parent.callbacks.updates).db
		if newDB.Error == nil && newDB.RowsAffected == 0 {
			return s.New().Table(scope.TableName()).FirstOrCreate(value)
		}
		return newDB
	}
	return scope.callCallbacks(s.parent.callbacks.creates).db
}
```
we won't be able to use this method, and I'm not sure how to implement this with `gorm` in that case,

and I don't think it is a big deal to warrant rewriting this code

Reply:
although https://github.com/decred/politeia/pull/1412#discussion_r633001987 should probably solve my consern regarding using `Save` method

Reply:
Same reasoning as this.

https://github.com/decred/politeia/pull/1412#discussion_r633118213